src/rust/engine/graph/src/node.rs

```diff
@@ -18,6 +19,8 @@ pub type EntryId = stable_graph::NodeIndex<u32>;
 ///
 /// Defines executing a cacheable/memoizable step within the given NodeContext.
 ///
+/// Note that it is assumed that Nodes are very cheap to clone.
```

Comment:
This seems reasonable given https://github.com/pantsbuild/pants/pull/6078

Reply:
Yep, thanks! This was a comment that I realized later that I should have added here.